packages/plugin-core/src/web/commands/lookup/LookupQuickpickFactory.ts

```diff
@@ -0,0 +1,269 @@
+import {
+  DNodeUtils,
+  DVault,
+  FuseEngine,
+  NoteLookupUtils,
+  NoteQuickInputV2,
+} from "@dendronhq/common-all";
+import _ from "lodash";
+import path from "path";
+import { inject, injectable } from "tsyringe";
+import * as vscode from "vscode";
+import { QuickPick, QuickPickOptions } from "vscode";
+import { type IReducedEngineAPIService } from "../../engine/IReducedEngineApiService";
+import { WSUtilsWeb } from "../../utils/WSUtils";
+import { type ILookupProvider } from "./ILookupProvider";
+import { VaultQuickPick } from "./VaultQuickPick";
+
+const CREATE_NEW_LABEL = "Create New";
+
+export type LookupQuickpickFactoryCreateOpts = QuickPickOptions & {
+  provider: ILookupProvider;
+  buttons?: vscode.QuickInputButton[];
+  initialValue?: string;
+};
+
+export type LookupAcceptPayload = {
+  items: readonly NoteQuickInputV2[];
+  createNew?: boolean;
+};
+
+@injectable()
+export class LookupQuickpickFactory {
+  private _engine: IReducedEngineAPIService;
+
+  constructor(
+    @inject("IReducedEngineAPIService") engine: IReducedEngineAPIService,
+    @inject("vaults") private vaults: DVault[],
+    private wsUtils: WSUtilsWeb
+  ) {
+    this._engine = engine;
+  }
+
+  public showLookup(
+    opts: LookupQuickpickFactoryCreateOpts
+  ): Promise<LookupAcceptPayload | undefined> {
+    let initialValue = opts?.initialValue;
+    if (!initialValue) {
+      initialValue = this.getInitialValueBasedOnActiveNote();
+    }
+
+    const qp = this.create({
+      title: "Lookup Note",
+      buttons: [],
+      provider: opts.provider,
+      initialValue,
+    });
+
+    const outerPromise = new Promise<LookupAcceptPayload | undefined>(
+      (outerResolve) => {
+        const foo = new Promise<LookupAcceptPayload | undefined>((resolve) => {
```

Comment:
I'd use a more meaningful variable name than foo lol

Reply:
good catch haha - there's a lot of cleanup I was planning on doing to this file.  I want to do it as part of implementing a Schema lookup controller so that I can better understand what needs to be abstracted.